improve serverless dms doc (#15793)
pingcap/docs: tidb-cloud/tidb-cloud-connect-aws-dms.md

Reviewer: preview: https://pingcap-docs-website-scaffold-git-preview-cloud-57b396-qiancai.vercel.app/tidbcloud/tidb-cloud-connect-aws-dms

## Prerequisites

### An AWS account with enough access

You are expected to have an AWS account with enough access to manage DMS-related resources. If not, refer to the following AWS documents:

- [Sign up for an AWS account](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_GettingStarted.SettingUp.html#sign-up-for-aws)
- [Identity and access management for AWS Database Migration Service](https://docs.aws.amazon.com/dms/latest/userguide/security-iam.html)

### A TiDB Cloud account and a TiDB cluster

You are expected to have a TiDB Cloud account and a TiDB Serverless or TiDB Dedicated cluster. If not, refer to the following documents to create one:

- [Create a TiDB Serverless cluster](/tidb-cloud/create-tidb-cluster-serverless.md)
- [Create a TiDB Dedicated cluster](/tidb-cloud/create-tidb-cluster.md)

## Network configuration

> Before creating DMS resources, you need to configure network properly to ensure DMS can communicate with TiDB Cloud clusters. If you are unfamiliar with AWS, contact AWS Support. We give several possible configurations here.

Reviewer: "If you are unfamiliar with AWS, contact AWS Support." 这里这么写不知道是否恰当，或者 If you are unfamiliar with AWS, see xxx documentation 之类的可以吗

- To connect to a TiDB Dedicated cluster via VPC peering, [set up a VPC peering connection](/tidb-cloud/set-up-vpc-peering-connections.md) first and deploy the replication instance to private subnets.
</div>
</SimpleTab>

## Create an AWS DMS replication instance

1. Go to the [Replication instances](https://console.aws.amazon.com/dms/v2/home#replicationInstances) page in the AWS DMS console, and switch to the corresponding region. It is recommended to use the same region for AWS DMS as TiDB Cloud.

   ![Create replication instance](/media/tidb-cloud/aws-dms-tidb-cloud/aws-dms-connect-1.png)

2. Click **Create replication instance**.

3. Fill in an instance name, ARN, and description.

4. Configure the instance:
    - **Instance class**: select an appropriate instance class. For more information, see [Choosing replication instance types](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_ReplicationInstance.Types.html).
    - **Engine version**: use the default configuration.
    - **High Availability**: select **Single-AZ** or **Multi-AZ** based on your business needs.

5. Configure the storage in the **Allocated storage (GiB)** field.

> 6. Configure connectivity and security. Check previous section for network configuration.

Reviewer: ```suggestion
6. Configure connectivity and security. You can check [the previous section](#network-configuration) for network configuration.
```

---
title: Connect AWS DMS to TiDB Cloud clusters

> summary: Learn how to migrate data from or into TiDB Cloud using AWS DMS.

Reviewer 2: ```suggestion
summary: Learn how to connect AWS DMS to TiDB Cloud clusters.
```

---
title: Connect AWS DMS to TiDB Cloud clusters
summary: Learn how to migrate data from or into TiDB Cloud using AWS DMS.
---

# Connect AWS DMS to TiDB Cloud clusters

[AWS Database Migration Service (AWS DMS)](https://aws.amazon.com/dms/) is a cloud service that makes it possible to migrate relational databases, data warehouses, NoSQL databases, and other types of data stores. You can use AWS DMS to migrate your data from or into TiDB Cloud clusters. This document describes how to connect AWS DMS to TiDB Cloud clusters.

## Prerequisites

### An AWS account with enough access

> You are expected to have an AWS account with enough access to manage DMS-related resources. If not, refer to the following AWS documents:

Author: It's the second link below.

- [Sign up for an AWS account](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_GettingStarted.SettingUp.html#sign-up-for-aws)
- [Identity and access management for AWS Database Migration Service](https://docs.aws.amazon.com/dms/latest/userguide/security-iam.html)

### A TiDB Cloud account and a TiDB cluster

You are expected to have a TiDB Cloud account and a TiDB Serverless or TiDB Dedicated cluster. If not, refer to the following documents to create one:

- [Create a TiDB Serverless cluster](/tidb-cloud/create-tidb-cluster-serverless.md)
- [Create a TiDB Dedicated cluster](/tidb-cloud/create-tidb-cluster.md)

## Network configuration

> Before creating DMS resources, you need to configure network properly to ensure DMS can communicate with TiDB Cloud clusters. If you are unfamiliar with AWS, contact AWS Support. We give several possible configurations here.

Reviewer: ```suggestion
Before creating DMS resources, you need to configure network properly to ensure DMS can communicate with TiDB Cloud clusters. If you are unfamiliar with AWS, contact AWS Support. The following provides several possible configurations for your reference.

## Create an AWS DMS replication instance

1. Go to the [Replication instances](https://console.aws.amazon.com/dms/v2/home#replicationInstances) page in the AWS DMS console, and switch to the corresponding region. It is recommended to use the same region for AWS DMS as TiDB Cloud.

   ![Create replication instance](/media/tidb-cloud/aws-dms-tidb-cloud/aws-dms-connect-1.png)

2. Click **Create replication instance**.

3. Fill in an instance name, ARN, and description.

4. Configure the instance:
    - **Instance class**: select an appropriate instance class. For more information, see [Choosing replication instance types](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_ReplicationInstance.Types.html).
    - **Engine version**: use the default configuration.
    - **High Availability**: select **Single-AZ** or **Multi-AZ** based on your business needs.

5. Configure the storage in the **Allocated storage (GiB)** field.

6. Configure connectivity and security. You can check [the previous section](#network-configuration) for network configuration.
    - **Network type - new**: select **IPv4**.
    - **Virtual private cloud (VPC) for IPv4**: select the VPC that you need.

>     - **Replication subnet group**: choose a subnet group for your replication instance.

Reviewer 3: ```suggestion
    - **Replication subnet group**: select a subnet group for your replication instance.
```

    - **Public accessible**: set it based on your network configuration.

    ![Connectivity and security](/media/tidb-cloud/aws-dms-tidb-cloud/aws-dms-connect-2.png)

> 7. Configure the **Advanced settings**, **Maintenance**, and **Tags** if needed, and then click **Create replication instance** to finish the instance creation.

Reviewer 3: ```suggestion
7. Configure the **Advanced settings**, **Maintenance**, and **Tags** sections if needed, and then click **Create replication instance** to finish the instance creation.
```

## Create TiDB Cloud DMS endpoints

For connectivity, the steps for using TiDB Cloud clusters as a source or as a target are similar, but DMS does have some different database setting requirements for source and target. For more information, see [Using MySQL as a source](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Source.MySQL.html) or [Using MySQL as a target](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Target.MySQL.html). When using a TiDB Cloud cluster as a source, you can only **Migrate existing data** because TiDB does not support MySQL binlog.

1. Go to the [Endpoints](https://console.aws.amazon.com/dms/v2/home#endpointList) page in the AWS DMS console, and switch to the corresponding region.

    ![Create endpoint](/media/tidb-cloud/aws-dms-tidb-cloud/aws-dms-connect-3.png)

2. Click **Create endpoint** to create the target database endpoint.

> 3. Select **Source endpoint** or **Target endpoint**.

Reviewer 3: ```suggestion
3. In the **Endpoint type** section, select **Source endpoint** or **Target endpoint**.
```

> 4. Fill in the **Endpoint identifier** and ARN, and then select **Source engine** or **Target engine** as **MySQL**.

Reviewer 3: ```suggestion
4. In the **Endpoint configuration** section, fill in the **Endpoint identifier** and ARN fields. Then, select **MySQL** as **Source engine** or **Target engine**.
```

> 5. Choose **Provide access information manually** and fill in cluster information:

Reviewer 3: ```suggestion
5. For the **Access to endpoint database** field, select the **Provide access information manually** checkbox and fill in cluster information as follows:
```

    <SimpleTab>

    <div label="TiDB Serverless">

    - **Server name**: `HOST` of TiDB Serverless cluster.
    - **Port**: `PORT` of TiDB Serverless cluster.
    - **User name**: User of TiDB Serverless cluster for migration. Make sure it meets DMS requirements.
    - **Password**: Password of the TiDB Serverless cluster user.
    - **Secure Socket Layer (SSL) mode**: If you are connecting via public endpoint, it is highly recommended to set the mode to **verify-full** to ensure transport security. If you are connecting via private endpoint, you can set the mode to **none**.

>     - **CA certificate**: [ISRG Root X1 certificate](https://letsencrypt.org/certs/isrgrootx1.pem). For more information, see [TLS Connections to TiDB Serverless](/tidb-cloud/secure-connections-to-serverless-clusters.md).

Reviewer 3: ```suggestion
    - (Optional) **CA certificate**: Use the [ISRG Root X1 certificate](https://letsencrypt.org/certs/isrgrootx1.pem). For more information, see [TLS Connections to TiDB Serverless](/tidb-cloud/secure-connections-to-serverless-clusters.md).
```

    </div>

    <div label="TiDB Dedicated">

    - **Server name**: `HOST` of TiDB Dedicated cluster.
    - **Port**: `PORT` of TiDB Dedicated cluster.
    - **User name**: User of TiDB Dedicated cluster for migration. Make sure it meets DMS requirements.
    - **Password**: Password of TiDB Dedicated cluster user.
    - **Secure Socket Layer (SSL) mode**: If you are connecting via public endpoint, we highly recommend setting it to **verify-full** to ensure transport security. If you are connecting via private endpoint, you can set it to **none**.

>     - **CA certificate**: Get the CA certificate according to [TLS Connections to TiDB Dedicated](/tidb-cloud/tidb-cloud-tls-connect-to-dedicated.md).

Reviewer 3: ```suggestion
    - (Optional) **CA certificate**: Get the CA certificate according to [TLS connections to TiDB Dedicated](/tidb-cloud/tidb-cloud-tls-connect-to-dedicated.md).
```

    </div>
    </SimpleTab>

     ![Provide access information manually](/media/tidb-cloud/aws-dms-tidb-cloud/aws-dms-connect-4.png)

> 6. If the endpoint to be created is a **Target endpoint**, set **Extra connection attributes** to `Initstmt=SET FOREIGN_KEY_CHECKS=0;`.

Reviewer 3: ```suggestion
6. If you create the endpoint as a **Target endpoint**, expand the **Endpoint settings** section, select the **Use endpoint connection attributes** checkbox, and then set **Extra connection attributes** to `Initstmt=SET FOREIGN_KEY_CHECKS=0;`.
```

It's hard to find the **Use endpoint connection attributes** field, because the **Endpoint settings** section is collapsed by default and we need to select **Use endpoint connection attributes** first.

> 7. Configure the **KMS Key** and **Tags** if needed. Click **Create endpoint** to finish the instance creation.

Reviewer 3: ```suggestion
7. Configure the **KMS Key** and **Tags** sections if needed. Click **Create endpoint** to finish the instance creation.
```

</div>
</SimpleTab>

## Create an AWS DMS replication instance

1. In the AWS DMS console, go to the [**Replication instances**](https://console.aws.amazon.com/dms/v2/home#replicationInstances) page and switch to the corresponding region. It is recommended to use the same region for AWS DMS as TiDB Cloud.

   ![Create replication instance](/media/tidb-cloud/aws-dms-tidb-cloud/aws-dms-connect-replication-instances.png)

2. Click **Create replication instance**.

3. Fill in an instance name, ARN, and description.

4. In the **Instance configuration** section, configure the instance:
    - **Instance class**: select an appropriate instance class. For more information, see [Choosing replication instance types](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_ReplicationInstance.Types.html).
    - **Engine version**: keep the default configuration.
    - **High Availability**: select **Multi-AZ** or **Single-AZ** based on your business needs.

5. Configure the storage in the **Allocated storage (GiB)** field.

> 6. Configure connectivity and security. You can refer to [the previous section](#network-configuration) for network configuration.

Reviewer 3: ```suggestion
6. Configure connectivity and security. You can refer to [the previous section](#configure-network) for network configuration.
```

    - (Optional) **CA certificate**: Use the [ISRG Root X1 certificate](https://letsencrypt.org/certs/isrgrootx1.pem). For more information, see [TLS Connections to TiDB Serverless](/tidb-cloud/secure-connections-to-serverless-clusters.md).

    </div>

    <div label="TiDB Dedicated">

    - **Server name**: `HOST` of TiDB Dedicated cluster.
    - **Port**: `PORT` of TiDB Dedicated cluster.
    - **User name**: User of TiDB Dedicated cluster for migration. Make sure it meets DMS requirements.
    - **Password**: Password of TiDB Dedicated cluster user.
    - **Secure Socket Layer (SSL) mode**: If you are connecting via public endpoint, it is highly recommended to set the mode to **verify-full** to ensure transport security. If you are connecting via private endpoint, you can set it to **none**.
    - (Optional) **CA certificate**: Get the CA certificate according to [TLS connections to TiDB Dedicated](/tidb-cloud/tidb-cloud-tls-connect-to-dedicated.md).

    </div>
    </SimpleTab>

     ![Provide access information manually](/media/tidb-cloud/aws-dms-tidb-cloud/aws-dms-connect-configure-endpoint.png)

> 6. If you create the endpoint as a **Target endpoint**, expand the **Endpoint settings** section, select the **Use endpoint connection attributes** checkbox, and then set **Extra connection attributes** to `Initstmt=SET FOREIGN_KEY_CHECKS=0;`.

Reviewer: ```suggestion
6. If you want to create the endpoint as a **Target endpoint**, expand the **Endpoint settings** section, select the **Use endpoint connection attributes** checkbox, and then set **Extra connection attributes** to `Initstmt=SET FOREIGN_KEY_CHECKS=0;`.
```
